core/src/test/resources/wiremock_integration/wiremock_with_template.json

```diff
@@ -0,0 +1,69 @@
+            "request": {
+                "mocks": [
+                    {
+                        "name": "Template 001",
+                        "operation": "GET",
+                        "request": {
+                            "urlPath": "/template/001"
+                        },
+                        "response": {
+                            "status": 200,
+                            "headers" : {
+                                "Content-Type" : "application/json"
+                            },
+                            "body": {
+                                // Showcases the use of WireMock's Response Templates
+                                // http://wiremock.org/docs/response-templating/
+                                // "fixed_date_time" is set by ZeroCode at mock-creation time, so it will remain the same accross subsequent calls
+                                // "dynamic_oneMinuteAgo", "dynamic_oneMinuteAhead" and "ucparam" are resolved at mock-response time, so they can be different per call
+                                // The templates are Handlebars Templates, where {{...}} delimits the templates, and inside you can put 'helpers'
+                                // now is one of those helpers that returns the current date (new java.util.Date()), and takes some extra attributes
+                                // to manipulate it. See the section 'Date and time helpers' on the WireMock-link for more info
+                                // upper is another helper, that uppercases it's argument (i.c. the first value of the request-parameter named 'param'
+                                "fixed_date_time": "${LOCAL.DATETIME.NOW:uuuu-MM-dd HH:mm:ss.SSS}",
+                                "dynamic_oneMinuteAgo": "{{now offset='-1 minutes' format='yyyy-MM-dd HH:mm:ss.SSS'}}",
+                                "dynamic_oneMinuteAhead": "{{now offset='+1 minutes' format='yyyy-MM-dd HH:mm:ss.SSS'}}",
+                                "ucparam": "{{upper request.query.param[0]}}"
+                            }
+                        }
+                    }
+                ]
+            },
+            "assertions": {
+                "status": 200
+            }
+        },
+        {
+            "name": "GetTemplate",
+            "url": "/template/001",
+            "operation": "GET",
+            "loop": 3,
+            "request": {
+                "headers": {
+                    "Content-Type": "application/json;charset=UTF-8"
+                },
+                "queryParams": {
+                    "param": "lowerToUpper"
+                },
+                "body": {
+                }
+            },
+            "assertions": {
+                "status": 200,
+                "body": {
+                    "ucparam": "LOWERTOUPPER"
+                    // Once PR#249 is accepted, "fixed_date_time": "$DATE.AFTER${$.GetTemplate.response.dynamic_oneMinuteAgo}"
```

Comment:
All other changes looks fine. But if you add this line, this test will fail.

correct from GetTemplate.response.dynamic_oneMinuteAgo to GetTemplate.response.body.dynamic_oneMinuteAgo and from $DATE.AFTER${} to $DATE.AFTER:${}


PR #249 is now merged. Now you can remove the comment & test it. I tried, But it's failing. Need to check.

Reply:
It is indeed failing, I realize now. Wiremock's 'now' returns a zoned datetime, while zerocode uses localdatetime everywhere: while parsing the zoned datetime, the zone is ignored.
E.g. where I'm living (Belgium), as response I get:
    "fixed_date_time" : "2019-06-15T14:03:53.211",
    "dynamic_oneMinuteAgo" : "2019-06-15T12:02:53Z",
    "dynamic_oneMinuteAhead" : "2019-06-15T12:04:54Z"

The first one is a local datetime, to be interpreted in the local timezone (+02:00), while the other two are expressed in UTC (+00:00) - 'Z' stands for 'Zero Timezone' (or, if your more adventurous: Zulu-timezone). The three are actually the same, apart from the minute-difference, but when parsed by LocalDateTime, the 'Z' is simply ignored. So the comparison sees a two-hour difference.

While I feel the best solution would be to switch from LocalDateTime to ZoneDateTime in ZeroCode, this would be a serious refactoring - which you, the ZeroCode-team, might have  valid reasons for rejecting. But I do wonder whether the use of LocalDateTime won't pose issues in real world usage of ZeroCode?

Two alternatives that I can safely implement for my use-case, both local to this PR:
- provide something like 'localnow' for use in WireMock response templates, that of course returns a LocalDateTime
- replace the fixed_date_time with the response of a JavaMethodExecution that returns a zoned datetime

I'll give the first one a try ...

Reply:
Thanks for the use case explanation. 👍 

Just trying to understand the usecase...

Basically, it sounds like if we can resolve the `fixed_date_time` field like below, then that might help to solve the usecase.
```
Something like below - Note the {LOCAL.DATETIME.NOW.Z: },

"fixed_date_time": "${LOCAL.DATETIME.NOW.Z:<date-time format>}",

=> resolving to
"fixed_date_time" : "2019-06-15T12:03:53.211Z", <- identical to wiremock date-time output
or
"fixed_date_time" : "2019-06-15T12:03:53Z", <- identical to wiremock date-time output
```

Is that correct assumption?
_(Then we can raise a new issue for this)_

_On the other note-_
Also, we are proposing to amend the earlier PR(#249 or via a new issue),
`$DATE.BEFORE` to `$LOCAL.DATETIME.BEFORE` to avoid confusion in the future.
Also same for : `$DATE.AFTER` to `$LOCAL.DATETIME.AFTER`
@jbonn360 could kindly help us here(happy to amend the issue and reopen)

Once we complete that, you need to amend the`assertion` part  in this PR to match this.
(_Apologies for this_)

Reply:
The solution would work, but it's not really correct: it would be outputting a zoned-datetime, but the markup says it's a *LOCAL*.DATETIME. 
And if the real problem (ignoring the timezone while parsing dates) would be solved, it just wouldn't matter anymore. Both local and zoned datetimes could be compared independently.

Reply:
ok. I am still trying to understand...
Does that mean would  a new markup help here?
e.g.
"fixed_date_time": "${ZONED.DATETIME.NOW:<date-time format>}",
or
You are good now and have already sorted out your usecase for now?

> Both local and zoned datetimes could be compared without the comparison needing to be aware of the original format.

Bit confused but this. 

Are you saying here, rather than solving the `ZONED.DATETIME` generation, better to create a solution around deserializing the timestamps. Which means - While comparing the date-time values, we should really have two options
1) For LocalDateTime
2) For ZonedDateTime
Then based on the format(or some way to distinguish them), parse the string to `LocalDateTime` or `ZonedDateTime`, then compare ?

Reply:
My usecase is sorted out now - no need to worry about that anymore.

And yes, that is what I wanted to say. We should honour the timezone if it is present, assume the system timezone if not. I think this would do the trick:

```
public static LocalDateTime parseDate(String value) {
     try {
         ZonedDateTime zonedDateTime = ZonedDateTime.parse(value, DateTimeFormatter.ISO_DATE_TIME);
         return zonedDateTime.withZoneSameInstant(ZoneId.systemDefault()).toLocalDateTime();
     } catch ( DateTimeParseException e ) {
         return LocalDateTime.parse(value, DateTimeFormatter.ISO_DATE_TIME);
     }
}
```